platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +251,242 @@ def get_sfp(self, index):
+        The down state will power down the module and the status should show
+        MODULE_STATUS_OFFLINE.
+        The up state will take the module to MODULE_STATUS_FAULT or
+        MODULE_STATUS_ONLINE states.
+
+        Args:
+            up: A boolean, True to set the admin-state to UP. False to set the
+            admin-state to DOWN.
+
+        Returns:
+            bool: True if the request has been issued successfully, False if not
+        """
+        if up:
+            power_on_success = self.dpuctl_obj.dpu_power_on()
+            self.fault_state = not power_on_success
+            return power_on_success
+        return self.dpuctl_obj.dpu_power_off()
+
+    def get_type(self):
+        """
+        Retrieves the type of the module.
+
+        Returns:
+            A string, the module-type from one of the predefined types:
+            MODULE_TYPE_SUPERVISOR, MODULE_TYPE_LINE or MODULE_TYPE_FABRIC
+            or MODULE_TYPE_DPU or MODULE_TYPE_SWITCH
+        """
+        return ModuleBase.MODULE_TYPE_DPU
+
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by SUPERVISOR, LINE-CARD,
+        FABRIC-CARD, SWITCH, DPU0, DPUX
+
+        Returns:
+            A string, the module name prefixed by one of MODULE_TYPE_SUPERVISOR,
+            MODULE_TYPE_LINE or MODULE_TYPE_FABRIC or MODULE_TYPE_DPU or
+            MODULE_TYPE_SWITCH and followed by a 0-based index.
+
+            Ex. A Chassis having 1 supervisor, 4 line-cards and 6 fabric-cards
+            can provide names SUPERVISOR0, LINE-CARD0 to LINE-CARD3,
+            FABRIC-CARD0 to FABRIC-CARD5.
+            A SmartSwitch having 4 DPUs and 1 Switch can provide names DPU0 to
+            DPU3 and SWITCH
+        """
+        return self._name
+
+    def get_description(self):
+        """
+        Retrieves the platform vendor's product description of the module
+
+        Returns:
+            A string, providing the vendor's product description of the module.
+        """
+        return "NVIDIA BlueField-3 DPU"
+
+    def get_oper_status(self):
+        states_ = [ModuleBase.MODULE_STATUS_FAULT,
+                   ModuleBase.MODULE_STATUS_OFFLINE,
+                   ModuleBase.MODULE_STATUS_ONLINE,
+                   ModuleBase.MODULE_STATUS_MIDPLANE_ONLINE,
+                   ModuleBase.MODULE_STATUS_CONTROLPLANE_ONLINE]
+        oper_state = states_[0]
+        if self.fault_state:
+            return states_[0]
+        elif utils.read_int_from_file(f'/run/hw-management/events/{self.get_name().lower()}_shtdn_ready') == 1:
+            return states_[1]
+        if utils.read_int_from_file(f'/run/hw-management/events/{self.get_name().lower()}_ready') == 1:
+            oper_state = states_[2]
+            if self.is_midplane_reachable():
+                oper_state = states_[3]
+                if self._is_dataplane_int_online():
+                    oper_state = states_[4]
```

Comment:
"If if if if if" constructions are not readable. Change this to the following:
```
if self.fault_state:
    return ModuleBase.MODULE_STATUS_FAULT

dataplane_state = self._is_dataplane_int_online()
midplane_state = self.is_midplane_reachable()

if dataplane_state and midplane_state:
    return ModuleBase.MODULE_STATUS_CONTROLPLANE_ONLINE

if dataplane_state:
     return ModuleBase.MODULE_STATUS_DATAPLANCE_ONLINE

if midplane_state:
    return ModuleBase.MODULE_STATUS_MIDPLANE_ONLINE

if self._is_online():
     return ModuleBase.MODULE_STATUS_ONLINE

return ModuleBase.MODULE_STATUS_OFFLINE
```

Reply:
Done